server/protocols/jmap/src/main/java/org/apache/james/jmap/utils/CollectionHierarchySorter.java

```diff
@@ -19,36 +19,46 @@
 
 package org.apache.james.jmap.utils;
 
+import java.util.Collection;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
-import org.apache.james.jmap.model.mailbox.Mailbox;
 import org.apache.james.jmap.utils.DependencyGraph.CycleDetectedException;
 
 import com.google.common.collect.Lists;
 
-public class MailboxHierarchySorter {
+public class CollectionHierarchySorter<T, Id> {
```

Comment:
Sorry to be that picky, but the naming does not "click".
May I suggest HierarchicalCollections as a broader helper or SortingHierarchicalCollections